src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -656,102 +577,71 @@ public void startSession(String path, String userName, String botName) throws IO
+      return null;
     }
 
-    // create a chat session from the bot.
-    Chat chat = new Chat(bot);
-    // load session specific predicates, these override the default ones.
-    String sessionPredicateFilename = createSessionPredicateFilename(userName, botName);
-    chat.predicates.getPredicateDefaults(sessionPredicateFilename);
+    session = new Session(this, userName, botInfo);
+    sessions.put(getSessionKey(userName, botName), session);
 
-    if (!sessions.containsKey(botName)) {
-      // initialize the sessions for this bot
-      HashMap<String, ChatData> newSet = new HashMap<String, ChatData>();
-      sessions.put(botName, newSet);
-    }
-    // put the current user session in the sessions map (overwrite if it was
-    // already there.
-    sessions.get(botName).put(userName, new ChatData(chat));
-    initializeChatSession(userName, botName, chat);
-    // this.currentBotName = botName;
-    log.info("Started session for bot name:{} , username:{}", botName, userName);
-    setReady(true);
-  }
-
-  private void initializeChatSession(String userName, String botName, Chat chat) {
-    // lets test if the robot knows the name of the person in the session
-    String name = chat.predicates.get("name").trim();
-    // TODO: this implies that the default value for "name" is default
-    // "Friend"
-    if (name == null || "Friend".equalsIgnoreCase(name) || "unknown".equalsIgnoreCase(name)) {
-      // TODO: find another interface that's simpler to use for this
-      // create a string that represents the predicates file
-      String inputPredicateStream = "name:" + userName;
-      // load those predicates
-      chat.predicates.getPredicateDefaultsFromInputStream(FileIO.toInputStream(inputPredicateStream));
-    }
-    // TODO move this inside high level :
-    // it is used to know the last username...
-    if (sessions.get(botName).containsKey("default") && !userName.equals("default")) {
-      setPredicate("default", "lastUsername", userName);
-      // robot surname is stored inside default.predicates, not inside
-      // system.prop
-      setPredicate(userName, "botname", getPredicate("default", "botname"));
-      try {
-        savePredicates();
-      } catch (IOException e) {
-        // TODO Auto-generated catch block
-        e.printStackTrace();
-      }
-    }
+    log.info("Started session for bot botName:{} , userName:{}", botName, userName);
+    setCurrentSession(userName, botName);
+    return session;
+  }
+
+  /**
+   * setting the current session is equivalent to setting current user name and
+   * current bot name
+   * 
+   * @param userName
+   * @param botName
+   */
+  public void setCurrentSession(String userName, String botName) {
+    setCurrentUserName(userName);
+    setCurrentBotName(botName);
   }
 
   public void addCategory(Category c) {
-    Bot bot = bots.get(getCurrentBotName().toLowerCase());
+    Bot bot = getBot(getCurrentBotName());

```

Comment:
getBot is marked as deprecated, you shouldn't be using it if you are deprecating it.